### PR TITLE
Support named Meters, use global package

### LINKED ISSUE #168: Implement "Named Tracers and Meters"
The "Named Tracers and Meters" RFC (https://github.com/open-telemetry/oteps/blob/master/text/0016-named-tracers.md) has been approved and was added to the specification (open-telemetry/opentelemetry-specification#264).

Please make sure to also update the documentation if necessary (tracer, meter creation).

OpenTelemetry language repositories now need to implement this mechanisms.

### LINKED ISSUE #216: Move global Meter and Tracer getter/setters into dedicated globals package
The proposal is that we move all globals out of the main `trace` and `metrics` package.  The global tracer and meter objects will be registered in another package, `opentelemtry`, along with the `Init()` method discussed in OTEP 0005.

Eliminating globals from the main packages allows globals package to auto-initialize by loading a plugin, for example; if we separate globals from these packages now, this can be a future option.  If we leave globals in the main packages, we cannot do this. It is impossible to auto-load a plugin that depends on the package that is loading it. When there are no globals in the packages themselves, it becomes possible for a binary plugin to to be loaded that implements the `Tracer` or `Meter` interfaces, at the first moment when the globals are referenced.

There is an old proof-of-concept showing how this could work in the future, provided that we separate globals from the main API packages now.  https://github.com/open-telemetry/opentelemetry-go/issues/52

### PR DESCRIPTION
**Pull request by jmacd**
Resolves #216 
Resolves #168 

This brings the metric API to the same place the trace API is w.r.t. globals and providers

### CODE REVIEW

File: global/global.go [hunk 1]
```diff
@@ -17,27 +17,53 @@ package global
...
-func SetTraceProvider(p trace.Provider) {
-	globalP.Store(globalProvider{p: p})
+func SetTraceProvider(tp trace.Provider) {
+	globalTracer.Store(traceProvider{tp: tp})
+}
+
+// MeterProvider returns the registered global trace provider.
+// If none is registered then an instance of NoopMeterProvider is returned.
+// Use the trace provider to create a named tracer. E.g.
+//     tracer := global.MeterProvider().GetMeterr("example.com/foo")
```
Comment (rghetia) [inline global/global.go]: s/trace/meter/

s/NoopMeterProvider/NoopProvider/

File: global/global.go [hunk 2]
```diff
@@ -17,27 +17,53 @@ package global
...
+func MeterProvider() metric.Provider {
+	if gp := globalMeter.Load(); gp != nil {
+		return gp.(meterProvider).mp
+	}
+	return metric.NoopProvider{}
+}
+
+// SetMeterProvider registers p as the global trace provider.
```
Comment (rghetia) [inline global/global.go]: s/trace/meter/

File: global/global.go (hunk 1 above)
Comment (jmacd) [reply global/global.go]: Fixed.

File: global/global.go (hunk 2 above)
Comment (jmacd) [reply global/global.go]: Fixed.

**Pull request merged**